cmd/juju/application/deploy.go

```diff
@@ -794,9 +794,41 @@ func (c *DeployCommand) maybeReadLocalBundle() (deployFn, error) {
 	}, nil
 }
 
-func (c *DeployCommand) maybeReadLocalCharm() (deployFn, error) {
+func (c *DeployCommand) maybeReadLocalCharm(apiRoot DeployAPI) (deployFn, error) {
+	//NOTE: Here we select the series using the algorithm defined by the
+	//series selector. This serves to override the algorithm found in
+	//`charmrepo.NewCharmAtPath` which is outdated (but must still be called
+	//since the code is coupled with path interpretation logic which needs
+	//to be factored out).
+
+	ch, err := charm.ReadCharm(c.CharmOrBundle)
+	var series string
+	if err != nil {
+		series = c.Series
+	} else {
+		modelCfg, err := getModelConfig(apiRoot)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		seriesSelector := seriesSelector{
+			seriesFlag:      c.Series,
+			supportedSeries: ch.Meta().Series,
+			force:           c.Force,
+			conf:            modelCfg,
+			fromBundle:      false,
+		}
+
+		series, err = seriesSelector.charmSeries()
+		if charm.IsUnsupportedSeriesError(err) {
+			return nil, errors.Errorf("%v. Use --force to deploy the charm anyway.", err)
+		} else if err != nil {
+			series = c.Series
```

Comment:
What is the use case here? When will charmSeries return an error that isn't an unsupported series error?

Does it still make sense to just use the CLI --series, or should we be telling the user?